kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/testplugins/ProtocolCounterFilter.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.testplugins;
+
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.kafka.common.header.internals.RecordHeader;
+import org.apache.kafka.common.message.ProduceRequestData;
+import org.apache.kafka.common.message.RequestHeaderData;
+import org.apache.kafka.common.message.ResponseHeaderData;
+import org.apache.kafka.common.protocol.ApiKeys;
+import org.apache.kafka.common.protocol.ApiMessage;
+
+import io.kroxylicious.proxy.filter.FilterContext;
+import io.kroxylicious.proxy.filter.RequestFilter;
+import io.kroxylicious.proxy.filter.RequestFilterResult;
+import io.kroxylicious.proxy.filter.ResponseFilter;
+import io.kroxylicious.proxy.filter.ResponseFilterResult;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class ProtocolCounterFilter implements
+        RequestFilter,
+        ResponseFilter {
+
+    private static @NonNull byte[] toBytes(int count) {
+        return new byte[]{
+                (byte) ((count & 0xFF000000) >> 24),
+                (byte) ((count & 0x00FF0000) >> 16),
+                (byte) ((count & 0x0000FF00) >> 8),
+                (byte) count,
+        };
+    }
```

Comment:
wonder if we should pull the byte[] serde methods we're using into a util class so it's easier to find a serde for int/String/boolean/?? as we find new usages for the HeaderInjector